opennlp-tools/src/main/java/opennlp/tools/formats/irishsentencebank/IrishSentenceBankDocument.java

```diff
@@ -0,0 +1,262 @@
+      if (!root.equalsIgnoreCase("sentences")) {
+        throw new IOException("Expected root node " + root);
+      }
+
+      NodeList nl = doc.getDocumentElement().getChildNodes();
+      for (int i = 0; i < nl.getLength(); i++) {
+        Node sentnode = nl.item(i);
+        if (sentnode.getNodeName().equals("sentence")) {
+          String src = sentnode.getAttributes().getNamedItem("source").getNodeValue();
+          String trans = "";
+          Map<Integer, String> toks = new HashMap<>();
+          Map<Integer, List<String>> flx = new HashMap<>();
+          List<Span> spans = new ArrayList<>();
+          NodeList sentnl = sentnode.getChildNodes();
+          int flexes = 1;
+          StringBuilder orig = new StringBuilder();
+
+          for (int j = 0; j < sentnl.getLength(); j++) {
+            final String name = sentnl.item(j).getNodeName();
+            if (name.equals("flex")) {
+              String slottmp = sentnl.item(j).getAttributes().getNamedItem("slot").getNodeValue();
+              Integer slot = Integer.parseInt(slottmp);
+              if (slot > flexes) {
+                flexes = slot;
+              }
+
+              if (flx.get(slot) == null) {
+                flx.put(slot, new ArrayList<>());
+              }
+
+              String tkn = sentnl.item(j).getAttributes().getNamedItem("lemma").getNodeValue();
+              flx.get(slot).add(tkn);
+            } else if (name.equals("translation")) {
+              trans = sentnl.item(j).getFirstChild().getTextContent();
+            } else if (name.equals("original")) {
+              int last = 0;
+              NodeList orignl = sentnl.item(j).getChildNodes();
+              for (int k = 0; k < orignl.getLength(); k++) {
+                if (orignl.item(k).getNodeName().equals("token")) {
+                  String tmp = orignl.item(k).getFirstChild().getTextContent();
+                  spans.add(new Span(last, last + tmp.length()));
+
+                  String slottmp = orignl.item(k).getAttributes().getNamedItem("slot").getNodeValue();
+                  Integer slot = Integer.parseInt(slottmp);
+                  if (slot > flexes) {
+                    flexes = slot;
+                  }
+
+                  toks.put(slot, tmp);
+                  orig.append(tmp);
+                  last += tmp.length();              
+                } else if (orignl.item(k).getNodeName().equals("#text")) {
+                  String tmp = orignl.item(k).getTextContent();
+                  orig.append(tmp);
+
+                  if (!" ".equals(tmp)) {
+                    spans.add(new Span(advanceLeft(tmp, last), advanceRight(tmp, last)));
+                  }
+
+                  last += tmp.length();
+                } else {
+                  throw new IOException("Unexpected node: " + orignl.item(k).getNodeName());
+                }
+              }
+            } else if (name.equals("#text") || name.equals("#comment")) {
```

Comment:
Replace this else-if-else block with 

`} else if (!sentnode.getNodeName().equals("#text") && !sentnode.getNodeName().equals("#comment")) {
          throw new IOException("Unexpected node: " + sentnode.getNodeName());
        }`